### ISSUE TITLE
Improve check for multiple junit-platform.properties files

### ISSUE THREAD

**Issue by sormuras** [junit-team/junit5#2207]
## Overview

Today, the following warning is generated...

https://github.com/junit-team/junit5/blob/e79dde3a30b146000743e754fc4c3355cfe70736/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherConfigurationParameters.java#L60

...and emitted even if the `junit-platform.properties` resource is only once physically present.

This false positive is due to using...

https://github.com/junit-team/junit5/blob/e79dde3a30b146000743e754fc4c3355cfe70736/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherConfigurationParameters.java#L55

...which may include duplicated elements when a custom class loader hierarchy is used. For details see the **"Implementation Requirements"** at: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassLoader.html#getResources(java.lang.String)

## Example

JDK 11, IntelliJ IDEA, in-module (white box) testing with...

```java
  @Test
  void getJUnitPlatformProperties() throws Exception {
    var loader = getClass().getClassLoader();
    System.out.println("loader = " + loader);
    System.out.println("loader.parent = " + loader.getParent());
    System.out.println("loader.parent.parent = " + loader.getParent().getParent());
    System.out.println();
    System.out.println("single: " + loader.getResource("junit-platform.properties"));
    System.out.println();
    var urls = Collections.list(loader.getResources("junit-platform.properties"));
    urls.forEach(url -> System.out.println("multi: " + url));
    System.out.println();
    var distinct = new HashSet<>(urls);
    distinct.forEach(url -> System.out.println("distinct: " + url));
  }
```
...emits:
```
loader = jdk.internal.loader.ClassLoaders$AppClassLoader@28864e92
loader.parent = jdk.internal.loader.ClassLoaders$PlatformClassLoader@e24ddd0
loader.parent.parent = null

single: file:/.../.idea/out/test/de.sormuras.bach/junit-platform.properties

multi: file:/.../.idea/out/test/de.sormuras.bach/junit-platform.properties
multi: file:/.../.idea/out/test/de.sormuras.bach/junit-platform.properties

distinct: file:/.../.idea/out/test/de.sormuras.bach/junit-platform.properties
```

## Deliverables

- [x] Prune duplicate elements before checking the amount of found resources

**Comment by sormuras** [comment]
Fixed for 5.6.1 -- cherry-picking https://github.com/junit-team/junit5/commit/828cdb3beb2ec0b35bdc012170c4b8f74ec33222 onto `master`.

**Comment by marcphilipp** [comment]
see https://github.com/junit-team/junit5/commit/828cdb3beb2ec0b35bdc012170c4b8f74ec33222#r37959524

**Comment by marcphilipp** [comment]
Resolved by 10ff91837e48e8e0af18d2d400ac9f9cda54ad70 and 921ab442489a5844686babe42dedfc091cc39d37.
